src/gt4py/next/program_processors/runners/dace_fieldview/gtir_dataflow.py

```diff
@@ -739,66 +733,149 @@ def _visit_map(self, node: gtir.FunCall) -> ValueExpr:
-        # by the `neighbors` visitor to fill the skip values in the neighbors list.
-        prev_reduce_identity = self.reduce_identity
-        self.reduce_identity = reduce_identity
-
-        try:
-            input_expr = self.visit(node.args[0])
-        finally:
-            # ensure that we leave the visitor in the same state as we entered
-            self.reduce_identity = prev_reduce_identity
-
-        assert isinstance(input_expr, MemletExpr | ValueExpr)
-        input_desc = input_expr.dc_node.desc(self.sdfg)
-        assert isinstance(input_desc, dace.data.Array)
-
-        if len(input_desc.shape) > 1:
-            assert isinstance(input_expr, MemletExpr)
-            ndims = len(input_desc.shape) - 1
-            # the axis to be reduced is always the last one, because `reduce` is supposed
-            # to operate on `ListType`
-            assert set(input_expr.subset.size()[0:ndims]) == {1}
-            reduce_axes = [ndims]
+        assert input_expr.local_offset is not None
+        connectivity = dace_utils.connectivity_identifier(input_expr.local_offset)
+        connectivity_node = self.state.add_access(connectivity)
+        connectivity_desc = connectivity_node.desc(self.sdfg)
+        connectivity_desc.transient = False
+
+        desc = input_expr.dc_node.desc(self.sdfg)
+        if isinstance(input_expr, MemletExpr):
+            local_dim_indices = [i for i, size in enumerate(input_expr.subset.size()) if size != 1]
         else:
-            reduce_axes = None
+            local_dim_indices = list(range(len(desc.shape)))
+
+        assert len(local_dim_indices) == 1
+        local_dim_index = local_dim_indices[0]
+        assert desc.shape[local_dim_index] == offset_provider.max_neighbors
+
+        # we lower the reduction map with WCR out memlet in a nested SDFG
+        nsdfg = dace.SDFG("reduce_with_skip_values")
+        nsdfg.add_array(
+            "values",
+            (desc.shape[local_dim_index],),
+            desc.dtype,
+            strides=(desc.strides[local_dim_index],),
+        )
+        nsdfg.add_array(
+            "neighbor_indices",
+            (connectivity_desc.shape[1],),
+            connectivity_desc.dtype,
+            strides=(connectivity_desc.strides[1],),
+        )
+        nsdfg.add_scalar("acc", desc.dtype)
+        st_init = nsdfg.add_state("init")
+        st_init.add_edge(
+            st_init.add_tasklet(
+                "init_acc",
+                {},
+                {"__val"},
+                f"__val = {reduce_init.dc_dtype}({reduce_init.value})",
+            ),
+            "__val",
+            st_init.add_access("acc"),
+            None,
+            dace.Memlet(data="acc", subset="0"),
+        )
+        st_reduce = nsdfg.add_state_after(st_init, "reduce")
+        st_reduce.add_mapped_tasklet(
+            name="reduce_with_skip_values",
+            map_ranges=dict(i=f"0:{offset_provider.max_neighbors}"),
```

Comment:
Everywhere else you are using `{}` except here.
```suggestion
            map_ranges={"i": f"0:{offset_provider.max_neighbors}"},
```

Reply:
Done.